### DISCUSSION TITLE
Question: working with large images with constrained bandwidth

### DISCUSSION THREAD

**Discussion by treved72** [CropperBlazor/Cropper.Blazor#386] [Q&A] [answered]
Hi,

Let me explain my use case. 

We have field technicians that need to go to remote areas and photograph equipment. Given the remoteness of some sites, the bandwidth can be very low.

We have a Blazor Server application they use via laptops to upload site reports. Various images that are uploaded need to be cropped, and we want to use the built in resizing functionality.

Looking at one of the sample applications
```
        <InputFile id="imageInput" OnChange="InputFileChange" accept="image/*" hidden multiple />
```
```
        public async Task InputFileChange(InputFileChangeEventArgs inputFileChangeEventArgs)
        {
            var imageFile = inputFileChangeEventArgs.File;
            if (imageFile != null)
            {
                var oldSrc = Src;
                Src = await cropperComponent!.GetImageUsingStreamingAsync(imageFile, imageFile.Size);
                IsErrorLoadImage = false;
                cropperComponent?.Destroy();
                cropperComponent?.RevokeObjectUrlAsync(oldSrc).AsTask();
            }
        }
```

My reading of this is that when an image is selected, it's sent to the server - is that correct?

Is there a way to do this all locally and only send the cropped images to the server?

**Comment by MaxymGorn** [comment, marked as answer]
@treved72 yes, image will be uploaded to server. In addition, you can verify it locally via Dev Tools in Network Tab
<img width="1280" alt="image" src="https://github.com/user-attachments/assets/c1d09664-367a-42d5-9dca-494f0585ff29">

Moreover, I recommend using the new features of prerender modes from .net 8 blazor and processing on the client side (maybe hybrid mode with client and server)
